tensorflow/lite/tools/versioning/op_version.cc

```diff
@@ -616,6 +637,29 @@ OpSignature GetOpSignature(const OperatorCode* op_code, const Operator* op,
       }
     } break;
 
+    case BuiltinOperator_ADD: {
+      auto add_option = op->builtin_options_as_AddOptions();
+      op_sig.options.addsub.pot_scale_int16 = false;
+      if (add_option) {
+        op_sig.options.addsub.pot_scale_int16 = add_option->pot_scale_int16();
+      }
+    } break;
+
+    case BuiltinOperator_SUB: {
+      auto sub_option = op->builtin_options_as_SubOptions();
+      op_sig.options.addsub.pot_scale_int16 = false;
+      if (sub_option) {
+        op_sig.options.addsub.pot_scale_int16 = sub_option->pot_scale_int16();
+      }
+
+      if (op_code->builtin_code() == BuiltinOperator_SUB) {
+        op_sig.options.broadcast.need_broadcast =
```

Comment:
Looking at this some more, I don't think this usage of a union is valid. That is, you're assigning values to different members of the same union. I think we'd need to add some broadcast fields to addsub.

Reply:
Sorry, but i think I didn't get it, honestly. 
broadcast is independent of addsub - it was like this before

Reply:
The problem is that op_sig.options is a union. So if you assign something in op_sig.options.addsub, then you assign something in op_sig.options.broadcast, you're referencing the same memory region, which leads to undefined behavior. So if you want the broadcast check for add/sub ops, you'll need to add num_dims/needs_broadcast in the addsub struct.